File: moviejava/app/src/main/java/com/example/moviejava/ui/person/PersonActivity.java
package com.example.moviejava.ui.person;

import android.graphics.Color;
import android.os.Bundle;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.GridLayoutManager;

import com.example.moviejava.R;
import com.example.moviejava.databinding.ActivityPersonBinding;
import com.example.moviejava.ui.base.BaseActivity;
import com.example.moviejava.ui.home.adapter.MovieAdapter;

import javax.inject.Inject;

import dagger.hilt.android.AndroidEntryPoint;


@AndroidEntryPoint
public class PersonActivity extends BaseActivity<ActivityPersonBinding> {


    private PersonViewModel viewModel;
    @Inject
    MovieAdapter movieAdapter;


    @Override
    protected ActivityPersonBinding getLayoutBinding() {
        return ActivityPersonBinding.inflate(getLayoutInflater());
    }

    @Override
    protected void onHandleObject() {
        setSupportActionBar(binding().toolbar);
        getSupportActionBar().setDisplayShowTitleEnabled(true);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setHomeAsUpIndicator(R.drawable.ic_baseline_arrow_back_24);
        viewModel = new ViewModelProvider(this).get(PersonViewModel.class);
        Bundle bundle = getIntent().getExtras();
        int id = bundle.getInt("person_id");
        viewModel.getData(id);
                binding().showMore.setShowingLine(4);
        binding().showMore.setShowLessTextColor(Color.parseColor("#FF42E2F6"));
        binding().showMore.setShowMoreTextColor(Color.parseColor("#FF42E2F6"));
        binding().showMore.addShowLessText("hide");
        binding().showMore.addShowMoreText("read more");
    }

    @Override
    protected void onHandleEvent() {
        defaultObserver(viewModel.person, null, null, null, it -> {

            binding().setP(it);
            binding().executePendingBindings();
        });
        defaultObserver(viewModel.movie, null, null, null, it -> movieAdapter.submitData(it));

        binding().rcvMovie.setAdapter(movieAdapter);
        binding().rcvMovie.setLayoutManager(new GridLayoutManager(this, 4));
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
            return true;
        } else {
            super.onOptionsItemSelected(item);
        }
        return super.onOptionsItemSelected(item);
    }

}

File: movie/app/src/main/java/com/example/movie/ui/person/PersonActivity.kt
package com.example.movie.ui.person

import android.graphics.Color
import android.os.Bundle
import android.view.MenuItem
import androidx.activity.viewModels
import androidx.recyclerview.widget.GridLayoutManager
import com.example.movie.R
import com.example.movie.databinding.ActivityPersonBinding
import com.example.movie.ui.base.BaseActivity
import com.example.movie.ui.home.adapter.MovieAdapter
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint
class PersonActivity : BaseActivity<ActivityPersonBinding>() {


    private val viewModel: PersonViewModel by viewModels()

    @Inject
    lateinit var movieAdapter: MovieAdapter

    override fun getLayoutBinding(): ActivityPersonBinding =
        ActivityPersonBinding.inflate(layoutInflater)

    override fun onHandleObject() {
        setSupportActionBar(binding.toolbar)
        supportActionBar?.run {
            setDisplayShowTitleEnabled(true)
            setDisplayHomeAsUpEnabled(true)
            setHomeAsUpIndicator(R.drawable.ic_baseline_arrow_back_24)
        }
        val bundle: Bundle? = intent.extras
        val id = bundle!!.getInt("person_id")
        viewModel.getPerson(id)
        binding.rcvMovie.apply {
            adapter=movieAdapter
            layoutManager = GridLayoutManager(this@PersonActivity, 4)
        }
        binding.showMore.apply {
            setShowingLine(4)
            setShowLessTextColor(Color.parseColor("#FF42E2F6"))
            setShowMoreTextColor(Color.parseColor("#FF42E2F6"))
            addShowLessText("hide")
            addShowMoreText("read more")

        }
    }

    override fun onHandleEvent() {
        viewModel.apply {
            defaultObserver(person, null, null) {
                binding.p = it
                binding.executePendingBindings()
            }
            defaultObserver(movie, null, null) {

                it?.let { movieAdapter.submitData(it) }

            }
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> true.also { finish() }
            else -> super.onOptionsItemSelected(item)
        }
        return super.onOptionsItemSelected(item)

    }
}